django/db/models/fields/__init__.py

```diff
@@ -516,17 +516,47 @@ def clone(self):
     def __eq__(self, other):
         # Needed for @total_ordering
         if isinstance(other, Field):
-            return self.creation_counter == other.creation_counter
+            return (
+                self.creation_counter == other.creation_counter and
+                hasattr(self, 'model') == hasattr(other, 'model') and
```

Comment:
I think this is redundant.

Reply:
It's a super corner case that may not need to be addressed, but I didn't feel comfortable deciding that. The corner case is where one has a model set to `None` explicitly, but the other has no model set at all. The comparison on the next line would view that as the same, so that difference would be missed without this line. It seems quite reasonable to me that this is a condition we shouldn't worry about.